fractal/client/client.py

```diff
@@ -77,6 +112,20 @@ async def project_new(name: str, path: str, dataset: str) -> None:
     saved.
     """
     from fractal.common.models import ProjectBase
+    from fractal.common.models import ProjectRead
+
+    # Verify that name is not already in use
+    async with httpx.AsyncClient() as client:
+        auth = AuthToken(client=client)
+        res = await client.get(
+            f"{settings.BASE_URL}/project/",
+            headers=await auth.header(),
+        )
+        data = res.json()
+        project_list = [ProjectRead(**item) for item in data]
+        project_names = [p.name for p in project_list]
+        if name in project_names:
+            raise Exception(f"Project with {name=} already exists")
```

Comment:
the normal way of handling this is that you submit the post without a client side check and verify that the server returns  a 201 CREATED status code. If this is not the case, you handle the exception.

Reply:
At the moment, removing that check always leads to `res.status_code` being `201`, even if I create two projects with the same name (which is the behavior I want to prevent).

I guess I should first tell db that it can only allow a given project name once, at a per-user basis.

Reply:
As per our discussion, I moved this check (per-user check of unique project.slug) to the server endpoint - see https://github.com/fractal-analytics-platform/fractal/pull/142/commits/4b5aeaa2b93f944d232afc86e567d56254c1e66f.

Other commits moved the `unique=True` flag from `common/models/__init__.py` to `server/app/models/project.py`, but this is a global (that is, non per-user) constraint.
I guess the correct thing to do is to just remove it, and enforce the check upon project creation (as already done). In this way two users may have projects with the same slug. Is this the way we want to proceed?

Reply:
After our follow-up discussion, I removed the `unique=True` constraint. At the moment the `slug` field is still there, but plays no role in uniqueness constraints.
I also modified the endpoint to better enforce the constraint of unique (user,project_name) pairs.